restaurant/src/Components/Register.js

```diff
@@ -11,17 +11,8 @@ const Register = ()=>{
     const [pass , setPass] = useState('');
     const [rePass , setRePass] = useState('');
     const [email , setEmail] = useState('');
-    const [person , setPerson] = useState(new Person('' , '' , '' , '' , ''));
     const [invalidRegister , setInvalidRegister] = useState({errNick : '*' , errFirstName : '*' , errLastName : '*' , errPass : '*' , errRePass : '*' , errEmail : '*'});
```

Comment:
This is not immediately obvious and it took me a while to realise what's going on. I would prefer something more granular, like a setState per property.

Plus, you are duplicating a lot of code in `<Register/>`. You could have a `<InputField/>` component that manages its own state and error logic. Register would pass a setMyValue hook to it via props, and that's how the inputField could update back the Register on changes.

Small learning session: Another thing to keep in mind, which is not avoidable even with the InputField approach mentioned above, is that whenever setState is called, __the parent and all its children are re-rendered__; careful how you update the state, and read on useEffect to understand how to set State conditionally. For example, a useEffect without dependencies triggers on __every render__, which at best leads to obvious infinite re-renders, at worst to silent, expensive fetches from backend. For a good old register form you should not worry about this.